tasks.php

```diff
@@ -649,7 +650,7 @@ function handle_action_on_a_specified_task()
         $action = 'show';
     }
 
-    $task_id = (int)get_float_param($_REQUEST, 'task_id', null, 1, null);
+    $task_id = get_integer_param($_REQUEST, 'task_id', null, 1, null);
```

Comment:
The original was weird and I went back to do a `git blame` and this was me with:
```
commit f43d4da633fdc66a665683d673f401f9d6b02b43
Author: Casey Peel <cpeel@users.noreply.github.com>
Date:   Mon Dec 24 05:19:59 2018 +0000

    Use type=number inputs for numeric form fields

    For a better user experience, use number field types in forms where the
    input is only numeric. This isn't perfect as some browsers will allow
    the user to input an 'e' (expotenetial notation) and floating point
    numbers are also allowed, but it's a step up from the wide-open
    anything-goes that it was prior.
```

So in 2018 when we converted the input field to a numeric some browsers were sending (or allowed?) a float instead of an int? I guess let's evaluate what, if anything, we need to do about this. It could have been that I was just trying to accept as many "valid but unusual" inputs for these fields, but lets confirm that.

My gut says your change is fine though.

Reply:
So I tested and we could still send a float value from the form right now. As you probably found out in 2018,  input allows any float value and doesn't prevent form submission if the `<input type="number">` is invalid.

The top answer on [this StackOverflow thread](https://stackoverflow.com/questions/37043867/how-to-avoid-decimal-values-in-input-type-number) would prevent this: some JS that blocks any non-digit input.

Reply:
Sounds good. I haven't added some of the solutions from the thread. We can always follow-up if we find that this is an issue with folks.

Reply:
I take the attitude that if someone is playing around with trying out things that aren't integers for task IDs, they deserve what they get. (Bearing in mind that I do that sort of thing to try to break various things in testing.)